inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -571,22 +571,40 @@ def normalize_journal_titles(obj, eng):
     Returns:
        None
     """
-    publications = obj.data.get('publication_info')
-
-    if not publications:
-        return None
+    publications = obj.data.get('publication_info', [])
 
     for index, publication in enumerate(publications):
-        if 'journal_title' in publication:
-            normalized_title = normalize_journal_title(publication['journal_title'])
+        normalized_title, journal_record = normalize_journal_title_entry(publication)
+        if normalized_title:
             obj.data['publication_info'][index]['journal_title'] = normalized_title
-            ref_query = RecordMetadata.query.filter(
-                RecordMetadata.json['_collections'].op('?')('Journals')).filter(
-                cast(RecordMetadata.json['short_title'], String) == type_coerce(normalized_title, JSON))
-            result = db.session.execute(ref_query).fetchone()
-
-            if result:
-                obj.data['publication_info'][index]['journal_record'] = result.records_metadata_json['self']
+        if journal_record:
+            obj.data['publication_info'][index]['journal_record'] = journal_record
+
+    references = obj.data.get("references", [])
+    for index, reference in enumerate(references):
+        publication_info = get_value(reference, 'reference.publication_info')
+        if not publication_info:
+            continue
+        normalized_title, journal_record = normalize_journal_title_entry(publication_info)
+        if normalized_title:
+            obj.data['references'][index]['reference']['publication_info']['journal_title'] = normalized_title
+        if journal_record:
+            obj.data['references'][index]['reference']['publication_info']['journal_record'] = journal_record
+
+
+def normalize_journal_title_entry(publication_info):
+    if 'journal_title' not in publication_info:
+        return None, None
+
+    normalized_title = normalize_journal_title(publication_info['journal_title'])
+    ref_query = RecordMetadata.query.filter(
+        RecordMetadata.json['_collections'].op('?')('Journals')).filter(
+        cast(RecordMetadata.json['short_title'], String) == type_coerce(normalized_title, JSON))
+    result = db.session.execute(ref_query).fetchone()
```

Comment:
could we do an http request on hep for that?

Reply:
This is different from task requirements and refinement. It will require new endpoint on HEP and task was estimated only for 3 points. If you want to do this kind of changes it should go as new task.

Reply:
I'm sorry the goal is not to use the db directly on next as much as possible, and this adds an extra dependency. Isn't it just a search request? ```q=_collections:Journals and short_title: bla```

Reply:
I know that goal is to not create new db queries on next but this code is not new code. It's moved from another function only.

Reply:
The reason for this is that indexing is async, so having the contents of the DB depend on stuff in ElasticSearch is inconsistent in principle. In practice it probably doesn't matter here, so I would be ok with doing the query on ES only and getting the record ID from there.

Reply:
So maybe we could create a task for this? We agreed not to change requirements when task is during review... :/